src/LinuxTracing/UprobesUnwindingVisitorTest.cpp

```diff
@@ -840,6 +843,97 @@ TEST_F(UprobesUnwindingVisitorTest,
   EXPECT_EQ(discarded_samples_in_uretprobes_counter, 0);
 }
 
+TEST_F(UprobesUnwindingVisitorTest,
+       VisitSingleFrameStackSampleInFunctionToStopAtSendsCompleteCallstackAndAddressInfos) {
+  StackSamplePerfEvent event = BuildFakeStackSamplePerfEvent();
+
+  absolute_address_to_size_of_functions_to_stop_at_[kTargetAddress1] = 100;
+
+  EXPECT_CALL(return_address_manager_, PatchSample).Times(1).WillOnce(Return());
+  EXPECT_CALL(maps_, Get).Times(1).WillOnce(Return(nullptr));
+
+  std::vector<unwindstack::FrameData> callstack{kFrame1};
+
+  EXPECT_CALL(unwinder_, Unwind(event.data.pid, nullptr, _, _, event.data.dyn_size, _, _))
+      .Times(1)
+      .WillOnce(Return(LibunwindstackResult{callstack, {}, unwindstack::ErrorCode::ERROR_NONE}));
+
+  orbit_grpc_protos::FullCallstackSample actual_callstack_sample;
+  EXPECT_CALL(listener_, OnCallstackSample).Times(1).WillOnce(SaveArg<0>(&actual_callstack_sample));
+
+  std::vector<orbit_grpc_protos::FullAddressInfo> actual_address_infos;
+  auto save_address_info =
+      [&actual_address_infos](orbit_grpc_protos::FullAddressInfo actual_address_info) {
+        actual_address_infos.push_back(std::move(actual_address_info));
+      };
+  EXPECT_CALL(listener_, OnAddressInfo).Times(1).WillRepeatedly(Invoke(save_address_info));
+
+  std::atomic<uint64_t> unwinding_errors = 0;
+  std::atomic<uint64_t> discarded_samples_in_uretprobes_counter = 0;
+  visitor_.SetUnwindErrorsAndDiscardedSamplesCounters(&unwinding_errors,
+                                                      &discarded_samples_in_uretprobes_counter);
+
+  PerfEvent{std::move(event)}.Accept(&visitor_);
+
+  EXPECT_THAT(actual_callstack_sample.callstack().pcs(), ElementsAre(kTargetAddress1));
+  EXPECT_EQ(actual_callstack_sample.callstack().type(), orbit_grpc_protos::Callstack::kComplete);
+  EXPECT_THAT(actual_address_infos,
+              UnorderedElementsAre(AllOf(
+                  Property(&orbit_grpc_protos::FullAddressInfo::absolute_address, kTargetAddress1),
+                  Property(&orbit_grpc_protos::FullAddressInfo::function_name, kFunctionName1),
+                  Property(&orbit_grpc_protos::FullAddressInfo::offset_in_function, 0),
+                  Property(&orbit_grpc_protos::FullAddressInfo::module_name, kTargetName))));
+
+  EXPECT_EQ(unwinding_errors, 0);
+  EXPECT_EQ(discarded_samples_in_uretprobes_counter, 0);
+}
+
+TEST_F(UprobesUnwindingVisitorTest,
+       VisitSingleFrameStackSampleOutsideAnyFunctionToStopSendsUnwindingErrorAndAddressInfos) {
```

Comment:
- ToStop -> ToStopAt
- OutsideAnyFunction -> OutsideOfAnyFunction (my bad from the previous review)

Reply:
Done